src/plugins/intel_gpu/src/runtime/ocl/ocl_memory.cpp

```diff
@@ -462,6 +470,10 @@ event::ptr gpu_usm::copy_from(stream& stream, const memory& other, bool blocking
 }
 
 event::ptr gpu_usm::copy_from(stream& stream, const void* host_ptr, bool blocking) {
+    if (_bytes_count == 0) {
```

Comment:
Just out of curiosity: shouldn't we implement the same condition for OpenCL buffers as well?

Reply:
Hmm my basic intention of removing this "0 byte memcpy" is for performance matter rather than the functionality, because AFAIK the driver actually should support 0 byte memcopy. So I thought it is just a "nice to have" feature, but for the consistency, I will apply it for that case too :)

Reply:
Oh, got it, I thought we were supposed to address this issue on our side, thanks for clarification